<launch>
    <arg name="spawn_manager" default="true"/>
    <arg name="output" default="screen"/>

    <param name="vaccum_description" textfile="$(find aleph_viz)/urdf/vaccum.urdf"/>

    <rosparam file="$(find aleph2_hardware_interface)/config/common.yaml" command="load"/>
    <rosparam file="$(find aleph2_hardware_interface)/config/vaccum/joints.yaml" command="load"/>
    <rosparam file="$(find aleph2_hardware_interface)/config/vaccum/controllers.yaml" command="load"/>
    <rosparam file="$(find aleph2_hardware_interface)/config/vaccum/joint_limits.yaml" command="load"/>

    <group if="$(arg spawn_manager)">
        <node name="controller_manager" pkg="aleph2_hardware_interface" type="controller_manager" output="$(arg output)" required="true">
            <remap from="aleph2" to="vaccum"/>
            <remap from="robot_description" to="vaccum_description"/>
        </node>
        <node name="state_controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="$(arg output)" args="
            /aleph2/controller/state
        "/>
    </group>

    <node name="vaccum_controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="$(arg output)" args="
        /vaccum/controller/position/vaccum_rotator
    "/>
</launch>